libs/core/synchronization/include/hpx/synchronization/range_mutex.hpp

```diff
@@ -25,89 +25,114 @@ namespace hpx::synchronization {
     template <typename RangeMutex>
     class range_guard
     {
-        std::reference_wrapper<RangeMutex> lockRef;
-        std::size_t lockId = 0;
+        std::reference_wrapper<RangeMutex> mutex_ref;
+        std::size_t lock_id = 0;
 
     public:
         range_guard(RangeMutex& lock, std::size_t begin, std::size_t end)
-          : lockRef(lock)
+          : mutex_ref(lock)
         {
-            lockId = lockRef.get().lock(begin, end);
+            lock_id = mutex_ref.get().lock(begin, end);
         }
         ~range_guard()
         {
-            lockRef.get().unlock(lockId);
+            mutex_ref.get().unlock(lock_id);
+        }
+
+        range_guard(range_guard<RangeMutex>&) = delete;
+        range_guard<RangeMutex>& operator=(range_guard<RangeMutex>&) = delete;
+
+        range_guard(range_guard<RangeMutex>&& rhs_lock)
+        {
+            mutex_ref.get().unlock(lock_id);
```

Comment:
`lock_id` was never initialized for the object that the argument is being moved to.